File: spring-boot/src/main/java/edu/berkeley/gpporgs/model/Address.java
package edu.berkeley.gpporgs.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "addresses")
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String street;
    private String city;
    private String state;
    private Integer zip;
    private String country;
}

File: client/src/app/model/address.ts
import { deepCopy } from 'deep-copy-ts';

export class Address {
  id: number = null;
  street: string = null;
  city: string = null;
  state: string = null;
  zip: string = null;
  country: string = null;

  clone(): Address {
    return Object.assign(new Address(), deepCopy(this));
  }
}
